backend/presentation/Viewsets/comment_view.py

```diff
@@ -0,0 +1,126 @@
+from django.shortcuts import get_object_or_404
+from presentation.Serializers.comment_serializer import CommentSerializer
+from rest_framework import viewsets, status
+from rest_framework.response import Response
+import uuid
+from urllib.parse import urlparse
+
+'''
+URL: ://service/author/{author_id}/posts/{post_id}/comments access
+GET get comments of the post
+POST if you post an object of “type”:”comment”, it will add your comment to the post
+'''
+
+'''
+Manual Test:
+POST:
+{"displayName": "Lara Croft","github": "http://github.com/laracroft","username":"LaraCroft","email": "lara@gmail.com","password": "lara1234"}
+
+'''
+def getAuthorIDFromRequestURL(request, id):
+    parsed_url = urlparse(request.build_absolute_uri())
+    host = '{url.scheme}://{url.hostname}:{url.port}'.format(
+        url=parsed_url)
+    author_id = f"{host}/author/{id}"
+    return author_id
+
+def getPostIDFromRequestURL(request, id):
+    post_id = f"/posts/{id}"
+    return post_id
+
+def getCommentIDFromRequestURL(request, id):
+    comment_id = f"/comments/{id}"
+    return comment_id
+
+class CommentViewSet(viewsets.ModelViewSet):
+    serializer_class = CommentSerializer
+    queryset = Comment.objects.all()
+    lookup_field = 'id'
+
+    # GET a list of comments of the post
+    def list(self, request, *args, **kwargs):
+        author_id = getAuthorIDFromRequestURL(
+            request, self.kwargs['author_id'])
+        post_id = getPostIDFromRequestURL(
+            request, self.kwargs['post_id'])
+        post_id = author_id + post_id
+        post = get_object_or_404(Post, id=post_id)
+        queryset = Comment.objects.filter(post=post)
+        if queryset.exists():
+            comments = Comment.objects.get(post=post)
+            # May have mistakes here, do we need to change comment model?
+            return Response({
+                'type': 'comment',
+                'items': comments.items
+            })
+        else:
+            Comment.objects.create(post=post)
+            return Response({
+                'type': 'comment',
+                'items': []
+            })    
+
+    # GET a single comment using comment_id
+    def retrieve(self, request, *args, **kwargs):
+        author_id = getAuthorIDFromRequestURL(
+            request, self.kwargs['author_id'])
+        post_id = getPostIDFromRequestURL(
+            request, self.kwargs['post_id'])
+        comment_id = getCommentIDFromRequestURL(
+            request, self.kwargs['comment_id'])
+        post_id = author_id + post_id
+        comment_id = post_id + comment_id
+        post = get_object_or_404(Post, id=post_id)
+        comments = get_object_or_404(Comment, post=post)
+        if comment_id in comments.items:
+            # Is there a problem about the response?
+            f = get_object_or_404(Comment, id=comment_id)
+            return Response({'exist': True})
+        else:
+            return Response({'exist': False}, 404)
```

Comment:
This function is used to check if a comment exists, I think this one is not required.